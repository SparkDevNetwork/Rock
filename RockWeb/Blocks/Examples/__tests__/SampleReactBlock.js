
import { mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";

import SampleReactBlock from "../SampleReactBlock.block";

it("has a default state of 1", () => {
  const wrapper = mount(<SampleReactBlock />);
  expect(wrapper.state().counter).toEqual(1);
});

it("can be set an initial state from props", () => {
  const wrapper = mount(<SampleReactBlock startingNumber={10} />);
  expect(wrapper.state().counter).toEqual(10);
});

it("has the ability to increment when clicking the increment button", () => {
  const wrapper = mount(<SampleReactBlock />);
  const initialNumber = wrapper.state().counter;
  const button = wrapper.find("[data-spec=\"increment\"]");
  button.simulate("click");
  expect(wrapper.state().counter).toEqual(initialNumber + 1);
});

it("has the ability to decrement when clicking the decrement button", () => {
  const wrapper = mount(<SampleReactBlock />);
  const initialNumber = wrapper.state().counter;
  const button = wrapper.find("[data-spec=\"decrement\"]");
  button.simulate("click");
  expect(wrapper.state().counter).toEqual(initialNumber - 1);
});

it("matches the expected UI structure", () => {
  const wrapper = mount(<SampleReactBlock />);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});
