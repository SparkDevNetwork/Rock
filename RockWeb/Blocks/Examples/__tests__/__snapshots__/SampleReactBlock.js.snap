exports[`test matches the expected UI structure 1`] = `
<SampleReactBlock>
  <div>
    <h5>
      Counter is 
      1
    </h5>
    <button
      data-spec="increment"
      onClick={[Function]}
      type="button">
      Up
    </button>
    <button
      data-spec="decrement"
      onClick={[Function]}
      type="button">
      Down
    </button>
    <div
      style={
        Object {
          "paddingTop": "10px",
        }
      }>
      <p>
        <em>
          This is markup created by a react component. It is initialized with state from the server. Open the 
          <a
            href="https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=enj">
            React Dev Tools
          </a>
           and check out the state (counter amount), initialProps (startingNumber), and layout.
        </em>
      </p>
    </div>
  </div>
</SampleReactBlock>
`;
